pytorch_lightning/loops/optimization/optimizer_loop.py

```diff
@@ -204,7 +204,9 @@ def on_run_start(  # type: ignore[override]
         self, batch: Any, optimizers: List[Tuple[int, Optimizer]], batch_idx: int
     ) -> None:
         self._batch_idx = batch_idx
-        self._indices, self._optimizers = zip(*optimizers)
```

Comment:
ah wait. I removed exactly for that reason. mypy does correctly say a tuple is returneed

`self._indices` and `self._optimizers` are annotated as `List[...]`.

Was the intention that they are tuples? Should I change the List annotations instead?

Reply:
This is how it would work:
```py
>>> x = [(1, "a"), (2, "b")]
>>> zip(*x)
Out[3]: <zip at 0x7fdc6bf36d00>
>>> k, l = zip(*x)
>>> k
>>> Out[5]: (1, 2)
>>> l
>>> Out[6]: ('a', 'b')
```
And it seems the output is tuple. So I guess that's why it didn't match the typing. We could change the annotation of optimizers since it is used internally only.

Reply:
Yes, the code was cleaner.